### PR TITLE
DSG-604-Date-picker-support-relative-validation

### PR DESCRIPTION
**Pull request by g8shen**
# Basics

## PR Type
- [ ] Bugfix
- [x] Feature
- [ ] Styling and layout changes
- [ ] Code style changes (formatting, local variables)
- [ ] Refactoring (no functional changes)
- [ ] Build or CI related changes
- [ ] Tests
- [ ] Other... Please describe:

## What changes does this PR introduce?

A new component in the modal to set min/max relative date configuration. Currently can select a number of days/week/year. And styling is matching formios.

## Screenshot of the affected changes

https://github.com/Nuvalence/dsgov-web/assets/39806680/e87b3a46-9bd3-4e1b-bcb3-798f7963bae3
https://github.com/Nuvalence/dsgov-web/assets/39806680/93355e74-8a34-4b20-8e74-7c382f9d3dfa

## Checklist

- [ ] I have updated the documentation accordingly
- [ ] I have confirmed all acceptance criteria defined by the story have been met
- [ ] I have confirmed that Sonar Qube findings for new code are reviewed and fixed, or will be whitelisted

### CODE REVIEW

**Review by jgnuvalence — APPROVED** [review]

File: libs/shared/ui/nuverial/src/lib/components/date-picker/date-picker.component.ts [hunk 1]
```diff
@@ -166,4 +195,31 @@ export class NuverialDatePickerComponent extends FormInputBaseDirective implemen
     const newDate = datePipe.transform(this.formControl.value, 'yyyy-MM-dd');
     this.formControl.setValue(newDate);
   }
+
+  private calculateRelativeDate(value: string, key: string) {
```
Comment (davidyoung77) [inline libs/shared/ui/nuverial/src/lib/components/date-picker/date-picker.component.ts:199]: @g8shen @jgnuvalence we need to make sure we handle this logic in the api also

**Review by davidyoung77 — COMMENTED** [review]
The code looks good , I want to make sure we know how we are going to validate this in the api, Lets sync on this with @jgnuvalence before merging

**Review by davidyoung77 — APPROVED** [review]
We synced on this and have the plan for the api updates

File: libs/shared/ui/nuverial/src/lib/components/date-picker/date-picker.component.ts (hunk 1 above)
Comment (davidyoung77) [reply libs/shared/ui/nuverial/src/lib/components/date-picker/date-picker.component.ts:199]: We have talked through this and decided we need to handle negative dates

File: libs/shared/feature/form-nuv/src/lib/components/relative-date-input/relative-date-input.component.ts [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+    const numberOfUnitControl = this.formGroup.get('numberOfUnit')?.value;
+
+    if (dateUnitControl && numberOfUnitControl) {
+      const key = `${String(numberOfUnitControl)}-${String(dateUnitControl)}`;
+      this.updateValue(key);
+    } else {
+      this.updateValue('noRelativeDate');
+    }
+  }
+  //flag
```
Comment (davidyoung77) [inline libs/shared/feature/form-nuv/src/lib/components/relative-date-input/relative-date-input.component.ts:79]: Not sure what this is???

**Review by davidyoung77 — APPROVED** [review]
Nice approved with a comment

File: libs/shared/feature/form-nuv/src/lib/components/relative-date-input/relative-date-input.component.ts (hunk 2 above)
Comment (g8shen) [reply libs/shared/feature/form-nuv/src/lib/components/relative-date-input/relative-date-input.component.ts:79]: Oops sorry I will remove this.

**Review by kevincheng0 — APPROVED** [review]

**Pull request merged**